### ISSUE TITLE
`YaruBackButton` alignment behavior is inconsistent

### ISSUE THREAD

**Issue by d-loose** [ubuntu/yaru.dart#891]
Depending on whether its style is `round` or not, the button is wrapped in a `Center()` widget (see #531). Consequently, https://github.com/ubuntu/app-center/pull/1622 lead to a misalignment in the App Center's search page:
![image](https://github.com/ubuntu/yaru.dart/assets/113362648/7db34c55-a31c-445f-9882-9c778de04534)

Of course, this can be fixed very easily, but I suppose it's worth making the behavior consistent in yaru :slightly_smiling_face:

**Comment by spydon** [comment]
Ideally, doesn't this look better?

![image](https://github.com/ubuntu/yaru.dart/assets/744771/0a68e046-1245-4b1f-8f4a-4cd1185b09f0)

**Comment by Feichtmeier** [comment]
It should be in the leading spot of the appbar, when the page is paned, IMO, just like in the yaru.dart example :) or in the firmware updater / settings app / musicpod / etc
Not to distract from the actual issue, but the back button placement was really questionable from the beginning, IMHO 🙏

**Comment by spydon** [comment]
> It should be in the leading spot of the appbar, when the page is paned, IMO, just like in the yaru.dart example

That sounds like I good idea, but I can't find that in the yaru example?
Can you send a screenshot?

**Comment by Feichtmeier** [comment]
> > It should be in the leading spot of the appbar, when the page is paned, IMO, just like in the yaru.dart example
> 
> That sounds like I good idea, but I can't find that in the yaru example?
> Can you send a screenshot?

I just had a local change with this back button in the _TitleBar but got confused by the many ways in app center to access the navkey and reverted it

It could look like in this picture in MusicPod, but then on the snap/deb pages the share and flag buttons need to be moved I suppose

![album_dark.png](https://github.com/ubuntu/yaru.dart/assets/15329494/171eac06-3355-4946-b02e-983f4cc7d8c2)

**Comment by spydon** [comment]
That looks a lot better indeed, what do you think @anasereijo?

**Comment by anasereijo** [comment]
Indeed! This can work well now that we have approved @Feichtmeier's changes to the layout of the side nav. I am happy to proceed with this, make sure it's present on all pages that can have the back button (search results, app page, etc).

**Comment by Feichtmeier** [comment]
The issue remains but when I read Taha Tesser correctly, this issue with flutters leading widget in app bars is fixed, I think we can removed this internal center here, I'll have a look
